sympy/vector/tests/test_coordsysrect.py

```diff
@@ -479,3 +482,35 @@ def test_translation_trans_equations():
            (d.x + 1, d.y + 1, d.z)
     assert d._translation_trans_equations(a, d._origin, d.base_scalars(), inverse=True) ==\
            (d.x - 1, d.y - 1, d.z)
+
+
+def test_transformation_functions():
+    C = CoordSys3D("C")
+    D = C.locate_new("D", 3 * C.i)
+    assert D.transformation_from_parent() == (C.x + 3, C.y, C.z)
```

Comment:
didn't we conclude that `D.x == C.x - 3`? Express replaces `C.x` with `D.x + 3` in line 492...

Reply:
If we have passive transformation, the `parent` coordinate system is shifted. Here is shifted in left direction, I mean `-3`.
So if we have `from_parent` we're gong to the right direction, `+3`.
It's my explanation. I might be wrong

Reply:
Well, my purpose is to make `transformation_from_parent` return the combination of the parent's base scalars such that will be equivalent to the child's base scalars. Maybe the name was a bit unfortunate.

The idea is that you can use the resulting tuple and replace the value for `D.x` in `express`.

Maybe we should rename it to something more intuitive, like `base_scalars_from_parent`.